File: apps/cli/commands/status.ts
/**
 * Status command - Check daemon status
 */
import { getDaemonStatus } from "../daemon-manager.js";

export async function statusDaemon(): Promise<void> {
  const status = await getDaemonStatus();

  if (!status.running) {
    console.log("Daemon is not running");
    console.log("Run 'console start' to start the daemon");
    process.exit(0);
  }

  console.log("Daemon is running");
  console.log(`PID: ${status.pid}`);
  console.log(`Uptime: ${status.uptime || "unknown"}`);
  console.log(`Server: http://${status.host}:${status.port}`);
}

File: apps/cli-go/internal/commands/status.go
// Status command - check daemon status. Port of apps/cli/commands/status.ts.
package commands

import (
	"fmt"

	"github.com/Adelodunpeter25/console/apps/cli-go/internal/daemon"
)

// StatusDaemon prints whether the daemon is running.
func StatusDaemon() error {
	status := daemon.GetStatus()
	if !status.Running {
		fmt.Println("Daemon is not running")
		fmt.Println("Run 'console start' to start the daemon")
		return nil
	}

	fmt.Println("Daemon is running")
	fmt.Printf("PID: %d\n", status.Pid)
	uptime := status.Uptime
	if uptime == "" {
		uptime = "unknown"
	}
	fmt.Printf("Uptime: %s\n", uptime)
	fmt.Printf("Server: http://%s:%s\n", status.Host, status.Port)
	return nil
}
